crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1021,13 +1037,149 @@ impl IndexerStore for PgIndexerStore {
     }
 
     fn persist_epoch(&self, data: &TemporaryEpochStore) -> Result<(), IndexerError> {
-        // TODO: create new partition on epoch change
-        self.partition_manager.advance_epoch(data.epoch_id + 1)
+        self.partition_manager
+            .advance_epoch(data.new_epoch.epoch as u64)?;
+        let mut pg_pool_conn = get_pg_pool_connection(&self.cp)?;
+        pg_pool_conn
+            .build_transaction()
+            .serializable()
+            .read_write()
+            .run(|conn| {
+                if let Some(last_epoch) = &data.last_epoch {
+                    diesel::insert_into(epochs::table)
+                        .values(last_epoch)
+                        .on_conflict(epochs::epoch)
+                        .do_update()
+                        .set((
+                            epochs::last_checkpoint_id.eq(excluded(epochs::last_checkpoint_id)),
```

Comment:
not in this PR, but same thing as object updates, would be great if we have ways to test/even better ensure these cover all columns ..